indico/modules/events/abstracts/controllers/abstract_list.py

```diff
@@ -107,12 +110,40 @@ def _process(self):
 
 class RHCreateAbstract(RHAbstractListBase):
     def _process(self):
-        abstract_form_class = make_abstract_form(self.event, session.user, notification_option=True, management=self.management)
+        is_invited = request.args.get('invited') == '1'
+        abstract_form_class = make_abstract_form(self.event, session.user, notification_option=True,
+                                                 management=self.management, invited=is_invited)
         form = abstract_form_class(event=self.event, management=self.management)
+        if is_invited:
+            del form.description
+            del form.submitted_contrib_type
+            del form.attachments
+            del form.send_notifications
+            del form.person_links
+
         if form.validate_on_submit():
+            if is_invited:
+                if form.data['users_with_no_account'] == 'existing':
+                    submitter = form.data.pop('submitter')
```

Comment:
popping from `form.data` doesn't do anything useful as it's generated whenever the property is accessed. assign `data = form.data` earlier if you want a dict you can pop from

Reply:
That way you can also stop deleting stuff from `form` because you can just remove it from `data` instead..